src/watchmaker/__init__.py

```diff
@@ -280,10 +281,25 @@ def __init__(self, arguments):
         # All remaining arguments are worker_args
         self.worker_args = self._get_worker_args(arguments, extra_arguments)
 
-        self.config, status_config = \
-            get_configs(self.system,
-                        self.worker_args,
-                        self.config_path)
+        self.config, status_config = get_configs(
+            self.system, self.worker_args, self.config_path
+        )
+
+        computer_name_pattern = (
+            self.config.get("salt", {})
+            .get("config", {})
+            .get("computer_name_pattern", None)
+        )
+
+        computer_name = self.config["salt"]["config"].get(
+            "computer_name", None
+        )
+        if computer_name and computer_name_pattern:
+            if not re.match(computer_name_pattern + r"\Z", computer_name):
+                raise WatchmakerError(
+                    "Computer name: %s does not match pattern %s"
+                    % (computer_name, computer_name_pattern)
+                )
```

Comment:
In watchmaker, the workers own the "validation" of their arguments. There is a [`before_install()`](https://github.com/plus3it/watchmaker/blob/e777d7b5b57185835c79bcd6b26703a28bdfb112/src/watchmaker/workers/salt.py#L180-L199) we use to perform this kind of check.